`test-bump-version` workflow fails on release branch

**Describe the bug**
When the `test-bump-version` workflow runs in a release branch which has made a minor version number bump, this error occurs:
```python
  File "/home/runner/work/darker/darker/release_tools/bump_version.py", line 352, in get_next_milestone_version
    raise RuntimeError(f"No milestone exists for a version later than {version}")
RuntimeError: No milestone exists for a version later than 1.7.0
Error: Process completed with exit code 1.
```
In this example, the workflow was run for the `release-1.6.0` branch. The dry run for `bump_version.py` increases the version number to `1.7.0` and expects a `1.7.1` milestone to exist in the GitHub repository so it can link to it in `README.rst`.

**To Reproduce**
1. Do a minor version bump to `1.7.0` in a `release-1.7.0` branch when the time comes
2. Make sure a `1.8.1` milestone doesn't yet exist
3. Let the `test-bump-version` workflow run.
4. Look at the build failure for the `test-bump-version` workflow.

**Expected behavior**
No workflow failure.

**Links**
The failing build [#533](/akaihola/darker/actions/runs/3738972381/jobs/6345644916)

**Possible alternative solutions**
- Always take care that milestones for the next minor and patch versions exist on the Darker GitHub repository.
- In the workflow, do a patch version bump instead of a minor one, and just make sure a milestone for the next _minor_ version exists.
- Only emit a warning in `bump_version.py` when `--dry-run` is enabled.
- Create missing milestones automatically using the GitHub API.
- ...other?

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -18,6 +18,7 @@ Fixed
 - Fix tests to work with Git older than version 2.28.x.
 - GitHub Action example now omits ``revision:`` since the commit range is obtained
   automatically.
+- ``test-bump-version`` workflow will now succeed also in a release branch.
 
 
 1.6.0_ - 2022-12-19

--- a/release_tools/bump_version.py
+++ b/release_tools/bump_version.py
@@ -23,6 +23,7 @@ import sys
 from datetime import date
 from pathlib import Path
 from typing import Dict, Match, Optional, Tuple
+from warnings import warn
 
 import click
 import requests
@@ -112,6 +113,7 @@ def bump_version(  # pylint: disable=too-many-locals
         increment_major,
         increment_minor,
         token,
+        dry_run,
     )
     for path_str, pattern_templates in PATTERNS.items():
         path = Path(path_str)
@@ -188,6 +190,7 @@ def get_replacements(
     increment_major: bool,
     increment_minor: bool,
     token: Optional[str],
+    dry_run: bool,
 ) -> Tuple[PatternDict, ReplacementDict, Version]:
     """Return search patterns and replacements for version numbers and milestones
 
@@ -199,13 +202,16 @@ def get_replacements(
     :param increment_major: `True` to increment the major version number
     :param increment_minor: `True` to increment the minor version number
     :param token: The GitHub access token to use, or `None` to use none
+    :param dry_run: `True` if running in dry-run mode
     :return: Patterns, replacements and the new version number
 
     """
     old_version = get_current_version()
     new_version = get_next_version(old_version, increment_major, increment_minor)
     milestone_numbers = get_milestone_numbers(token)
-    next_version = get_next_milestone_version(new_version, milestone_numbers)
+    next_version = get_next_milestone_version(new_version, milestone_numbers, dry_run)
+    if dry_run:
+        milestone_numbers.setdefault(next_version, "MISSING_MILESTONE")
     patterns: PatternDict = {
         "any_version": r"\d+(?:\.\d+)*",
         "old_version": re.escape(str(old_version)),
@@ -342,12 +348,13 @@ def lookup_patterns(
 
 
 def get_next_milestone_version(
-    version: Version, milestone_numbers: Dict[Version, str]
+    version: Version, milestone_numbers: Dict[Version, str], dry_run: bool
 ) -> Version:
     """Get the next larger version number found among milestone names
 
     :param version: The version number to search a larger one for
     :param milestone_numbers: Milestone names and numbers from the GitHub API
+    :param dry_run: `True` if running in dry-run mode
     :return: The next larger version number found
     :raises RuntimeError: Raised if no larger version number could be found
 
@@ -355,7 +362,11 @@ def get_next_milestone_version(
     for milestone_version in sorted(milestone_numbers):
         if milestone_version > version:
             return milestone_version
-    raise RuntimeError(f"No milestone exists for a version later than {version}")
+    message = f"No milestone exists for a version later than {version}"
+    if not dry_run:
+        raise RuntimeError(message)
+    warn(message)
+    return Version(f"{version.major}.{version.minor}.{version.micro + 1}")
 
 
 def replace_span(span: Tuple[int, int], replacement: str, content: str) -> str:

